### PR TITLE
Create a Join Method on the Core Classes

### PR DESCRIPTION
**Pull request by Atomator**
Created a `join` method on the core class that allows for one core to be joined with another along a specified axis. In order for this operation to be performed, the `pixel_array` dimensions of the core that _aren't_ the joined axis must match with each other and each core must have the same `pixel_dimensions`.

### CODE REVIEW

File: src/core_ct/core.py [hunk 1]
```diff
@@ -282,3 +282,39 @@ def chunk(self, x1=0, y1=0, z1=0, x2=None, y2=None, z2=None) -> Core:
...
+            core: the `Core` object to join with the current core
+            axis: the axis to join the cores on
+
+        Returns:
+        -------
+            New core object made up of the two joined arrays
+
+        Raises:
+        ------
+            Exception if axis is a value other than 0, 1, or 2
```
Comment (carlameii) [inline src/core_ct/core.py:301]: Potentially change these from `Exception` to `ValueError` to be more specific.

File: src/core_ct/core.py [hunk 2]
```diff
@@ -282,3 +282,39 @@ def chunk(self, x1=0, y1=0, z1=0, x2=None, y2=None, z2=None) -> Core:
...
         return new_core
+
+    def join(self, core: Core, axis: int = 0) -> Core:
+        """
+        Join a core to the current core on specified axis.
+
+        Arguments:
+        ---------
+            core: the `Core` object to join with the current core
+            axis: the axis to join the cores on
```
Comment (carlameii) [inline src/core_ct/core.py:293]: I think we should mention that `axis` is an integer either 0, 1, or 2

**Review by carlameii — COMMENTED** [review]
I left a couple comments on the docstring, but otherwise looks good.

File: src/core_ct/core.py (hunk 1 above)
Comment (Atomator) [reply src/core_ct/core.py]: Updated all the `Exeception`s to be `ValueError`s

File: src/core_ct/core.py [hunk 3]
```diff
@@ -290,17 +290,20 @@ def join(self, core: Core, axis: int = 0) -> Core:
         Arguments:
         ---------
             core: the `Core` object to join with the current core
-            axis: the axis to join the cores on
+            axis: integer specifying which axis to rotate about (0, 1, or 2)
```
Comment (carlameii) [inline src/core_ct/core.py:293]: "rotate" should be changed to "join"

**Review by carlameii — APPROVED** [review]

**Review by KiraSqueak — APPROVED** [review]
Looks good to me

**Pull request merged**